overwatch/processing/alarms/collectors.py

```diff
@@ -19,54 +22,155 @@ class Singleton(_Singleton('SingletonMeta', (object,), {})):
+
+    Args:
+        alarmsParameters (dict): Parameters read from configuration files
+    Attributes:
+        sc (SlackClient):
+        channel (str): Channel name
+    """
     def __init__(self, alarmsParameters=None):
         if alarmsParameters is not None:
-            self.sc = SlackClient(alarmsParameters["apiToken"])
-            self.channel = alarmsParameters["slackChannel"]
+            self.parameters = alarmsParameters
+            # if 'slack' in self.parameters:
+            try:
+                self.sc = SlackClient(alarmsParameters["slack"]["apiToken"])
+                self.channel = alarmsParameters["slack"]["slackChannel"]
+            except KeyError:
+                logger.debug("Slack not configured")
 
     def __call__(self, alarm):
         self.sendMessage(alarm)
         
     def sendMessage(self, payload):
-        self.sc.api_call('chat.postMessage', channel=self.channel,
-                text=payload, username='Alarms OVERWATCH',
-                icon_emoji=':robot_face:')
+        """ Sends message to specified earlier channel.
+
+        Args:
+            payload (str): Message to send
+        Return:
+            None.
+        """
+        success = "Message successfully sent on Slack channel {channel}"
+        fail = "Slack not configured, couldn't send messages"
+
+        if 'slack' in self.parameters:
+            self.sc.api_call('chat.postMessage', channel=self.channel,
+                    text=payload, username='Alarms OVERWATCH',
+                    icon_emoji=':robot_face:')
+            logger.debug(success.format(channel=self.channel))
+        else:
+            logger.debug(fail)
+
+class AlarmCollector():
+    """
+    Class that collects generated alarms. Collected alarms are grouped and announced to
+    specified receivers.
+
+    Attributes:
+        alarms (list): List of alarms. Each element is a pair [Alarm, str]
+    """
+    def __init__(self):
+        self.alarms = []
+
+    def addAlarm(self, alarm):
+        """ It adds alarm to the existing list of alarms
+
+        Args:
+            alarm ([Alarm, str]): A pair - Alarm object and message
+        Return:
+            None.
+        """
+        self.alarms.append(alarm)
+
+    def announceAlarm(self):
+        """ It sends collected and grouped messages to receivers.
+        Then resets list of alarms. It can be called anywhere:
+        after processing each histogram, after each RUN, ect.
+
+        Args:
+            None.
+        Return:
+            None.
+        """
+        receivers = self._groupAlarms()
+        for receiver in receivers:
+            msg = '\n'.join(receivers[receiver])
+            receiver(msg)
+        self._resetCollector()
+
+    def _resetCollector(self):
+        self.alarms = []
+
+    def _groupAlarms(self):
+        receivers = {}
+        for alarmMsg in self.alarms:
+            alarm = alarmMsg[0]
+            msg = alarmMsg[1]
+            for receiver in alarm.receivers:
```

Comment:
I think that we should register alarm type to collector, and filter which alarms we want ot group